src/include/duckdb/common/types.hpp

```diff
@@ -365,7 +365,8 @@ enum class LogicalTypeId : uint8_t {
 	LIST = 101,
 	MAP = 102,
 	TABLE = 103,
-	ENUM = 104
+	ENUM = 104,
+	JSON = 105
```

Comment:
Since this is adding a new logical type, could we add it to the `test_all_types` function (`src/function/table/system/test_all_types.cpp`)?